src/service/timer-impl.js

```diff
@@ -78,13 +81,44 @@ export class Timer {
     }
     const wrapped = () => {
       try {
+        if (this.canceled_[id]) {
+          delete this.canceled_[id];
+          return;
+        }
         callback();
       } catch (e) {
         reportError(e);
         throw e;
       }
     };
-    return this.win.setTimeout(wrapped, opt_delay);
+
+    const delayEnd = Date.now() + opt_delay;
+    if (!this.delayQueue_[delayEnd]) {
+      this.delayQueue_[delayEnd] = [];
+      id = this.win.setTimeout(
```

Comment:
We can't return this id, since if the user calls `cancelTimeout` with it it would prevent all `cb`s registered for the same time from firing.

Reply:
Oh right